src/test/java/io/strimzi/kafka/metrics/KafkaMetricsCollectorTest.java

```diff
@@ -51,28 +50,27 @@ public void testCollect() {
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
 
-        // Adding a non-numeric metric does nothing
-        collector.addMetric(buildNonNumericMetric("name2", "group"));
-        metrics = collector.collect();
-        assertTrue(metrics.isEmpty());
-
         // Adding a metric that matches the allowlist
         collector.addMetric(buildMetric("name", "group", 1.0));
         metrics = collector.collect();
+        Collector.MetricFamilySamples getMetrics = metrics.get(0);
+
         assertEquals(1, metrics.size());
-        assertEquals("kafka_server_group_name", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(1.0, metrics.get(0).samples.get(0).value, 0.1);
-        assertEquals(new ArrayList<>(labels.keySet()), metrics.get(0).samples.get(0).labelNames);
-        assertEquals(new ArrayList<>(labels.values()), metrics.get(0).samples.get(0).labelValues);
+        assertEquals("kafka_server_group_name", getMetrics.name);
+        assertEquals(1, getMetrics.samples.size());
+        assertEquals(1.0, getMetrics.samples.get(0).value, 0.1);
```

Comment:
Sorry, I applied my comment yesterday to the wrong variable.  I meant to say extract `metrics.samples.get(0)` as a variable, say `serverGroupNameSamples`.   Looking again now, I actually think I'd expect a method assertMetric(..) which would validate that the expected metric exists, with the right value and labels.

Reply:
Addresses but would like feedback on whether it could be done better